lib/tessel/name.js

```diff
@@ -102,6 +102,11 @@ Tessel.prototype.rename = function(opts, callback) {
   else {
     self.getName(function(err, oldName) {
       if (err) return callback && callback(err);
+      // Don't set the new name if it's the same as the old name
+      if (oldName == opts.newName) {
+        tessel.logs.info("Name of device is already", oldName);
```

Comment:
I recommend that you just return this string as the first argument of the callback so it gets printed like an error:
```
callback && callback("Name of device is already " + oldName);
```